hw/ip/i2c/data/i2c.hjson

```diff
@@ -664,19 +679,39 @@
     }
     { name: "TIMEOUT_CTRL"
       desc: '''
-            I2C clock stretching timeout control.
-            This is used in I2C host mode to detect whether a connected target is stretching for too long.
+            I2C clock stretching and bus timeout control.
+            The behavior of this feature depends on the value of TIMEOUT_CTRL.BUS_TIMEOUT_EN.
+            If the bus timeout is disabled, this is used in I2C controller mode to detect whether a connected target is stretching a single low time beyond the timeout value.
+            When the bus timeout feature is disabled, this feature is more informative and doesn't do more than assert the "stretch_timeout" interrupt.
+
+            If the bus timeout feature is enabled, it is used to detect whether the clock has been held low for too long instead, inclusive of the controller's clock low time.
+            In an SMBus context, the VAL programmed should be tTIMEOUT:MIN, and EN and BUS_TIMEOUT_EN should be 1.
```

Comment:
This feature isn't totally complete yet, since the target FSM's responses aren't in this PR. Those will come in a follow-up.